Extending password entry leads to panic when password revealer is clicked

### Describe the bug:
I want users to be able to submit a username and password with a keypress (enter/return). Following the excellent tutorial on [handling keypresses](https://developer.fyne.io/tutorial/keypress-on-entry), I was able to do so. However, if the user clicks the password revealer, the app panics due to a segfault:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x41a8d2f]

goroutine 18 [running]:
fyne.io/fyne/v2/internal/app.(*FocusManager).Focus(0x0, 0x4409d00, 0xc000083080, 0x0)
	/Users/nick/go/src/fyne.io/fyne/v2/internal/app/focus_manager.go:25 +0x4f
fyne.io/fyne/v2/internal/driver/glfw.(*glCanvas).Focus(0xc0000b0000, 0x4409d00, 0xc000083080)
	/Users/nick/go/src/fyne.io/fyne/v2/internal/driver/glfw/canvas.go:88 +0x196
fyne.io/fyne/v2/widget.(*passwordRevealer).Tapped(0xc00008e280, 0xc0005748e0)
	/Users/nick/go/src/fyne.io/fyne/v2/widget/entry_password.go:49 +0xc8
fyne.io/fyne/v2/internal/driver/glfw.(*window).mouseClicked.func7()
	/Users/nick/go/src/fyne.io/fyne/v2/internal/driver/glfw/window.go:791 +0x38
fyne.io/fyne/v2/internal/driver/glfw.(*window).runEventQueue(0xc0000a2000)
	/Users/nick/go/src/fyne.io/fyne/v2/internal/driver/glfw/window.go:1242 +0x7a
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).createWindow.func1
	/Users/nick/go/src/fyne.io/fyne/v2/internal/driver/glfw/window.go:1266 +0x10d
```
It seems that embedding and using an entry with the password revealer causes it to panic on click. I am not sure if this also happens for other `ActionItem`s; I have only tested it in this one case.

### To Reproduce:
Run the example code and click each password revealer. The first two (which are not extended/embedded) work fine, but the second two (which are extended/embedded) both cause a panic on click.

### Screenshots:
![Screen Shot 2021-02-26 at 2 27 29 PM](https://user-images.githubusercontent.com/68124509/109361926-df06c380-783e-11eb-95a5-2996ae5605e3.png)

### Example code:
In this example code I don't add any functions to the extended entry -- in the real app there would be a method to handle key presses.
```go
package main

import (
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/container"
	"fyne.io/fyne/v2/widget"
)

type myPasswordEntry struct {
	widget.Entry
}

func newMyPasswordEntry() *myPasswordEntry {
	entry := &myPasswordEntry{
		widget.Entry{
			Password: true,
		},
	}
	entry.ExtendBaseWidget(entry)
	return entry
}

type myOtherPasswordEntry struct {
	*widget.Entry
}

func newMyOtherPasswordEntry() *myOtherPasswordEntry {
	entry := &myOtherPasswordEntry{
		&widget.Entry{
			Password: true,
		},
	}
	entry.ExtendBaseWidget(entry)
	return entry
}

func main() {
	myApp := app.New()
	window := myApp.NewWindow("Example")

	passwordEntry0 := widget.NewPasswordEntry()
	passwordEntry1 := &widget.Entry{Password: true}
	passwordEntry2 := newMyPasswordEntry()
	passwordEntry3 := newMyOtherPasswordEntry()

	passwordEntry0.SetPlaceHolder("Password revealer works")
	passwordEntry1.SetPlaceHolder("This one works too")
	passwordEntry2.SetPlaceHolder("This one crashes")
	passwordEntry3.SetPlaceHolder("This one crashes as well")

	vBox := container.NewVBox(passwordEntry0, passwordEntry1, passwordEntry2, passwordEntry3)

	window.SetContent(vBox)

	window.ShowAndRun()
}
```

### Device (please complete the following information):
 - **OS:** MacOS
 - **Version:** 11.2.1
 - **Go version:** 1.15.7
 - **Fyne version:** 2.0.0

--- a/widget/entry_password.go
+++ b/widget/entry_password.go
@@ -46,7 +46,7 @@ func (r *passwordRevealer) Tapped(*fyne.PointEvent) {
 	r.entry.setFieldsAndRefresh(func() {
 		r.entry.Password = !r.entry.Password
 	})
-	fyne.CurrentApp().Driver().CanvasForObject(r).Focus(r.entry)
+	fyne.CurrentApp().Driver().CanvasForObject(r).Focus(r.entry.super().(fyne.Focusable))
 }
 
 var _ fyne.WidgetRenderer = (*passwordRevealerRenderer)(nil)

--- a/widget/entry_password_extend_test.go
+++ b/widget/entry_password_extend_test.go
@@ -1,0 +1,31 @@
+package widget
+
+import (
+	"testing"
+
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/test"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type extendEntry struct {
+	Entry
+}
+
+func TestEntry_Password_Extended_CreateRenderer(t *testing.T) {
+	entry := &extendEntry{}
+	entry.ExtendBaseWidget(entry)
+	entry.Password = true
+	entry.Wrapping = fyne.TextTruncate
+	assert.NotNil(t, test.WidgetRenderer(entry))
+	r := test.WidgetRenderer(entry).(*entryRenderer).scroll.Content.(*entryContent)
+	p := test.WidgetRenderer(r).(*entryContentRenderer).provider
+	texts := test.WidgetRenderer(p).(*textRenderer).texts
+
+	test.Type(entry, "Pass")
+	assert.Equal(t, passwordChar+passwordChar+passwordChar+passwordChar, texts[0].Text)
+	assert.NotNil(t, entry.ActionItem)
+	test.Tap(entry.ActionItem.(*passwordRevealer))
+	assert.Equal(t, "Pass", texts[0].Text)
+}

